src/Containers/MiddleCanvas/DualSelectRadio.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import styled from 'styled-components'
+import { AngleDown } from '@styled-icons/fa-solid/AngleDown';
+import { ITemplatePrefill } from './MiddleCanvasTypes';
+import { RadioOptions } from './RadioOptions';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+import { Mixins } from '../../Utils';
+
+export interface DualSelectRadioProps extends MainInterface, ResponsiveInterface, React.HTMLAttributes<HTMLDivElement> {
+    caption?: string,
+    leftPlaceholder?: string,
+    rightPlaceholder?: string,
+    headerSpacingStyle?: string,
+    dualSelectOptions: ITemplatePrefill
+};
+
+const FIRST_SELECT_OPTION = 0;
+const SECOND_SELECT_OPTION = 1;
+const FIRST_LABEL = 0;
+
+export const DualSelectRadio: React.FC<DualSelectRadioProps> = ({
+    caption = 'Dual Select',
+    leftPlaceholder = 'Light/Sound',
+    headerSpacingStyle = 'space-between',
+    dualSelectOptions,
+    ...props
+}): React.ReactElement => {
+    const firstSelectOption = Object.values(dualSelectOptions)[FIRST_SELECT_OPTION].title;
+    const [leftSelectOption, setLeftSelectOption] = useState(Object.values(dualSelectOptions)[FIRST_SELECT_OPTION].labels[FIRST_LABEL]);
+    const [rightSelectOption, setRightSelectOption] = useState(Object.values(dualSelectOptions)[SECOND_SELECT_OPTION].labels[FIRST_LABEL]);
+    const [isCollapsed, setIsCollapsed] = useState(false);
```

Comment:
Object.values three times on the same prop doesn't seem right. Is there a better way to initialize this component?

Reply:
I've tried to think of other ways to do so, but I think it's difficult since I need to control:
1. which text it changes in the select area (Flash Once/Beep Once) without the first line I would have no way to tell whether I should be changing flash or beep
2. the second and third one individually control the select options of the light and sound options and they do not rely on each other

Would you have any suggestions to reduce the lines of code here?

Reply:
Just put an onPress on the buttons, so you know which option is selected

Reply:
If you let the component trigger the action you will know which it is